src/io/fasta.rs

```diff
@@ -70,7 +70,16 @@ impl<R: io::Read> Reader<R> {
         if !self.line.starts_with('>') {
             return Err(io::Error::new(io::ErrorKind::Other, "Expected > at record start."));
         }
-        record.header.push_str(&self.line);
+        record.id = match self.line[1..].trim_right().splitn(2, ' ').nth(0) {
+            Some("") => "".to_owned(),
```

Comment:
This line is superfluous, isn't it? It is covered by the id case below. Further, it would be better to use Option::map here.